src/components/CustomButton.vue

```diff
@@ -0,0 +1,108 @@
+<script setup>
+import { defineProps, defineEmits } from "vue";
+const props = defineProps({
+  iconSource: {
+    type: String,
+    default: "",
+  },
+  iconAlternativeText: {
+    type: String,
+    default: "",
+  },
+  label: {
+    type: String,
+    default: "",
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+defineEmits({
+  click: {},
+});
+
+function showIcon() {
+  return props.iconSource || props.iconAlternativeText;
+}
+
+function showLabel() {
+  return props.label;
+}
+
+function isDisabled() {
+  return props.disabled;
+}
+</script>
+
+<template>
+  <button
+    v-if="isDisabled"
+    :disabled="isDisabled()"
```

Comment:
No need for v-if? What is that even? It is always true since it is string??? 
```suggestion
    :disabled="disabled"
```